app/src/main/java/agency/five/tmdb/MainScreen.kt

```diff
@@ -0,0 +1,218 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.material.icons.filled.Home
+import androidx.compose.material.icons.outlined.Favorite
+import androidx.compose.material.icons.outlined.Home
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+
+@Composable
+fun MainScreen(
+    navController: NavController
+) {
+
+    // movies
+    val m0 = MovieItemViewState(
+        id = 0,
+        overview = "Nothing to show.",
+        title = "Iron Man",
+        imageResId = R.drawable.iron_man_1
+    )
+
+    val m1 = MovieItemViewState(
+        id = 1,
+        overview = "Nothing to show.",
+        title = "Gattaca",
+        imageResId = R.drawable.gattaca
+    )
+
+    val m2 = MovieItemViewState(
+        id = 2,
+        overview = "Nothing to show.",
+        title = "Lion King",
+        imageResId = R.drawable.lion_king
+    )
+
+    val m3 = MovieItemViewState(
+        id = 3,
+        overview = "Nothing to show.",
+        title = "Puppy Love",
+        imageResId = R.drawable.puppy_love
+    )
+
+    val m4 = MovieItemViewState(
+        id = 4,
+        overview = "Nothing to show.",
+        title = "Godzila",
+        imageResId = R.drawable.godzila
+    )
+
+    val m5 = MovieItemViewState(
+        id = 5,
+        overview = "Nothing to show.",
+        title = "Jungle Beat",
+        imageResId = R.drawable.jungle_beat
+    )
+
+
+    //list of movies to be shown on different sections tabs
+    val l1 = listOf(m0, m1, m2, m3)
+    val l2 = listOf(m0, m1)
+    val l3 = listOf(m4, m5)
+
+    // map of movies for the "What's popular" section with keys as tab names
+    val map1 = mapOf(
+        stringResource(id = R.string.streaming_tab) to l1,
+        stringResource(id = R.string.on_tv_tab) to l2,
+        stringResource(id = R.string.for_rent_tab) to l3,
+        stringResource(id = R.string.in_theaters_tab) to l1
+    )
+
+    // map of movies for the "Free to watch" section with keys as tab names
+    val map2 = mapOf(
+        stringResource(id = R.string.movies_tab) to l2,
+        stringResource(id = R.string.tv_tab) to l2
+    )
+
+    // map of movies for the "Trending" section with keys as tab names
+    val map3 = mapOf(
+        stringResource(id = R.string.today_tab) to l1,
+        stringResource(id = R.string.this_week_tab) to l3
+    )
+
+
+
+    var home by remember { mutableStateOf(true) }
```

Comment:
This is one way to track what tab is selected, but since you implemented the navigation between main and details screen with jetpack compose navigation, I suggest that you try to do the same for tabs.

That means that you'll have two NavHosts and two NavHostControllers
- One host for Main Screen and Details screen
- One host for Home tab and Favorites tab

I have attached an example of how you could structure the navigation graph. You'll probably have to change it slightly to work with your app, but the idea is implemented there. Let me know if you need more help with that. Notice that I have the same graph as you for root screens, but added one more for tabs.

```
package com.example.tmdb

// Imports
...

@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class)
@Composable
fun RootNavigationGraph(
    rootNavHostController: NavHostController
) {
    NavHost(rootNavHostController, startDestination = RootScreen.Main.route) {
        composable(RootScreen.Main.route) {
            MainScreen(rootNavHostController)
        }
        composable(
            route = Details.route + "/{${Details.ARGUMENT_ID}}",
            arguments = listOf(navArgument(Details.ARGUMENT_ID) { type = NavType.IntType })
        ) { entry ->
            DetailsScreen(rootNavHostController, mediaId = entry.arguments?.getInt(Details.ARGUMENT_ID))
        }
    }
}

sealed class RootScreen(
    val route: String,
    val title: String
) {
    object Main : RootScreen(
        route = "main",
        title = "Main"
    )

    object Details : RootScreen(
        route = "details_screen",
        title = "Details"
    ) {

        const val ARGUMENT_ID = "id"
    }
}

@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class)
@Composable
fun BottomBarNavigationGraph(
    rootNavHostController: NavHostController,
    bottomBarNavHostController: NavHostController
) {
    NavHost(bottomBarNavHostController, startDestination = BottomBarScreen.Home.route) {
        composable(BottomBarScreen.Home.route) {
            HomeScreen(rootNavHostController)
        }
        composable(BottomBarScreen.Favourites.route) {
            FavouritesScreen(rootNavHostController)
        }
    }
}

sealed class BottomBarScreen(
    val route: String,
    val title: String,
    val icon: Int,
) {
    object Home : BottomBarScreen(
        route = "home",
        title = "Home",
        icon = R.drawable.ic_home
    )

    object Favourites : BottomBarScreen(
        route = "favourites",
        title = "Favourites",
        icon = R.drawable.ic_favorite
    )
}
```